golos.publication/golos.publication.cpp

```diff
@@ -267,6 +191,47 @@ void publication::close_post_timer() {
     trx.send(_self, _self);
 }
 
+void publication::set_vote(account_name voter, account_name author,
+                           std::string permlink, asset weight,
+                           std::vector<structures::rshares> rshares) {
+    tables::vote_table vote_table(_self, author);
+
+    structures::post posttable_obj;
+
+    if (get_post(author, permlink, posttable_obj)) {
+        auto votetable_index = vote_table.get_index<N(postid)>();
+        auto votetable_obj = votetable_index.find(posttable_obj.id);
+
+        while (votetable_obj != votetable_index.end()) {
```

Comment:
why loop? there only one vote (from voter) can exist on any post/comment

Reply:
because can be a few same post_id